packages/protocol/contracts/governance/DowntimeSlasherSlots.sol

```diff
@@ -0,0 +1,351 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./SlasherUtil.sol";
+
+contract DowntimeSlasherSlots is SlasherUtil {
+  using SafeMath for uint256;
+
+  // Intervals previously slashed
+  struct SlashedInterval {
+    // startBlock of the slashed interval
+    uint256 startBlock;
+    // endBlock of the slashed interval. Although this could be calculated, as the downtime interval
+    // could be changed, we store the calculated end to avoid future errors if the interval changes.
+    uint256 endBlock;
+  }
+
+  // Accumulated ParentSealBitmap of every validator
+  struct ValidatedSlot {
+    // endBlock of the slashed interval. Although this could be calculated, as the downtime interval
+    // could be changed, we store the calculated end to avoid future errors if the interval changes.
+    uint256 endBlock;
+    // Accumulated ParentSealBitmap of the Slot
+    // The array will have 2 elements if the slot shares two epochs
+    uint256[2] validatorsUpAccumulator;
```

Comment:
One alternative I might suggest is to define slots such that they cannot cross epoch boundaries, which would allow you to guarantee that any slot only have one accumulator.